youtube_dl/extractor/generic.py

```diff
@@ -2532,6 +2532,18 @@ def _real_extract(self, url):
         if svt_url:
             return self.url_result(svt_url, 'SVT')
 
+        # Look for Squarespace sites' videos
+        matches = re.findall(r'<div class="[^"]*?sqs-block-video[^>]*>', webpage)
+        if matches:
+            sqs_urls = set()
+            for m in matches:
+                sqs_url = re.search(r'<iframe[^>]+?src=[^"]*"(//[^\\]+)', unescapeHTML(m))
+                if sqs_url:
+                    sqs_urls.add(sqs_url.group(1))
+            if sqs_urls:
+                return self.playlist_from_matches(
+                    sqs_urls, video_id, video_title)
```

Comment:
One more time: you must unescape and expand embed code **in place** leaving the rest to the generic extractor. You **must not** extract iframes on your own.

Reply:
Thanks, @dstftw!
I've updated, and the new code should be pretty close to what you're looking for.

And I've tested again, using the three problem sites, and all again download with the new code.